R/utils.R

```diff
@@ -340,3 +340,24 @@ flatten_dots <- function(...) {
   args <- rlang::dots_list(!!!args, .homonyms = "error")
   args
 }
+
+intersect_with_attr <- function(.l, ...) {
+  list <- c(.l, list(...))
+  attr_list <- purrr::map(list, attr_list)
+
+  intersect <- purrr::reduce(attr_list, intersect)
+  indices <- match(intersect, attr_list[[1]])
+  list[[1]][indices]
```

Comment:
This also has a bit of a smell: it's probably the constant `1`, which makes me think this function is doing something _very_ specific. OTOH `intersect_with_attr()` sounds generic enough that I think I understand what such a function would do, but I wouldn't expect these lines.